File: bai_tap_lon/lib/models/vocabulary.dart
class Vocabulary {
  final String id;
  final String word;
  final String meaning;
  final String phonetic;
  final String example;
  final String imageUrl;
  final String icon;

  Vocabulary({
    required this.id,
    required this.word,
    required this.meaning,
    required this.phonetic,
    required this.example,
    required this.imageUrl,
    this.icon = 'auto',
  });
}

File: be/app/Models/Vocabulary.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class Vocabulary extends Model
{
    protected $fillable = [
        'word',
        'phonetic',
        'meaning',
        'examples',
        'image_url',
        'icon',
    ];

    protected $casts = [
        'examples' => 'array',
    ];
}
